app/src/main/kotlin/com/wire/android/notification/NotificationManager.kt

```diff
@@ -0,0 +1,217 @@
+                        ),
+                        false,
+                        System.currentTimeMillis()
+                    )
+                )
+            )
+        )
+    }
+
+    fun testIt2() {
+        showNotification(
+            NotificationData(
+                listOf(
+                    NotificationConversation(
+                        "123412313",
+                        "Test User 2",
+                        null,
+                        listOf(
+                            NotificationMessage.Text("Sender1", "https://www.google.com/"),
+                        ),
+                        true,
+                        System.currentTimeMillis()
+                    ),
+                )
+            )
+        )
+    }
+
+    fun showNotification(data: NotificationData) {
+        val manager = NotificationManagerCompat.from(context)
+
+        createNotificationChannelIfNeeded(manager)
+        showSummaryIfNeeded(manager, data)
+        data.conversations.forEach { showConversationNotification(manager, it) }
+    }
+
+    private fun createNotificationChannelIfNeeded(manager: NotificationManagerCompat) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val notificationChannel = NotificationChannel(
+                CHANNEL_ID,
+                CHANNEL_NAME,
+                NotificationManager.IMPORTANCE_DEFAULT
+            )
+
+            manager.createNotificationChannel(notificationChannel)
+        }
+    }
+
+    private fun showSummaryIfNeeded(manager: NotificationManagerCompat, data: NotificationData) {
+        if (data.conversations.size <= 1) return
+
+        val summaryNotification = NotificationCompat.Builder(context, CHANNEL_ID)
+            .setSmallIcon(R.drawable.notification_icon_small)
+            .setGroup(GROUP_KEY)
+            .setGroupSummary(true)
+            .build()
+
+        manager.notify(SUMMARY_ID, summaryNotification)
+    }
+
+    private fun showConversationNotification(manager: NotificationManagerCompat, conversation: NotificationConversation) {
+        manager.notify(conversation.id.hashCode(), getConversationNotification(conversation))
+    }
+
+    private fun getConversationNotification(conversation: NotificationConversation) =
+        NotificationCompat.Builder(context.applicationContext, CHANNEL_ID).apply {
+            setSmallIcon(R.drawable.notification_icon_small)
+            setGroup(GROUP_KEY)
+            setAutoCancel(true)
+
+            setContentIntent(getPendingIntent(conversation.id))
+
+            setWhen(conversation.lastMessageTime)
+
+            setLargeIcon(conversation.image?.toBitmap())
+
+            addAction(getActionCall(conversation))
+            addAction(getActionReply(conversation.id))
+
+            val body = getFormattedText(conversation.messages, !conversation.isDirectChat)
+            setContentText(body)
+            setStyle(NotificationCompat.BigTextStyle().bigText(body))
```

Comment:
I think we need to use [MessagingStyle](https://developer.android.com/training/notify-user/expanded#message-style) here. It's created for messaging apps

Reply:
i was thinking about it. But i didn't find a way to hide user avatar there. 
In MessagingStyle by default, the message has UserIcon, UserName and MessageBody, but in our [design](https://www.figma.com/file/UgKehvwIbiIiQ2CCEdsVuf/Android?node-id=5272%3A92311) there is no UserImage at the left side (only `LargeIcon` on the right side).

Reply:
I think adapting the designs for the use of this `MessagingStyle` wouldn't be a problem

Reply:
I talked to Wolfgang and he approved changed design :)